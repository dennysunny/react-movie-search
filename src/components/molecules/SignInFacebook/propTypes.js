import { array } from 'prop-types';

export const propTypes = {
  history: array.isRequired,
};

export const defaultProps = {
};
