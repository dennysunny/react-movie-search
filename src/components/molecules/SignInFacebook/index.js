import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { Button } from 'react-bootstrap';

import { auth } from '../../../firebase';
import { propTypes } from './propTypes';

class SignInFacebook extends Component {
  static propTypes = propTypes;

  signInWithFB = () => {
    const { history } = this.props;

    auth.doSignInWithFacebook()
      .then(() => history.push('/'))
      // eslint-disable-next-line no-console
      .catch(err => console.log(err));
  }

  render() {
    return (
      <div>
        <Button
          type="submit"
          className="btn btn-primary btn-block"
          onClick={this.signInWithFB}
        >
          Log in with Facebook
        </Button>
      </div>
    );
  }
}

export default withRouter(SignInFacebook);
