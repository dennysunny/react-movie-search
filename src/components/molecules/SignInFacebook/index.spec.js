import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from 'react-bootstrap';
import { BrowserRouter } from 'react-router-dom';

import SignInFacebook from '.';

const mockProps = {
  history: [],
};

jest.mock('../../../firebase/auth', () => ({
  doSignInWithFacebook: () => new Promise(() => ({
    data: { user: { name: 'test' } },
  })),
}));

// jest.mock('../../../firebase/auth', () => ({
//   auth: jest.fn(() => ({
//     doSignInWithFacebook: () => ({
//       data: { user: { name: 'test' } },
//     }),
//   })),
// }));

describe('Molecules/SignInFacebook', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <SignInFacebook.WrappedComponent {...mockProps} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle signin button click', () => {
    const wrapper = mount(<SignInFacebook.WrappedComponent {...mockProps} />);
    wrapper.find(Button).simulate('click');
    expect(wrapper.props().history).toEqual([]);
  });
});
