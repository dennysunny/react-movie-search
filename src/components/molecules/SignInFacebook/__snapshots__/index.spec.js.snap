// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/SignInFacebook should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInFacebook
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn btn-primary btn-block"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Log in with Facebook
      </Button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "block": false,
        "bsClass": "btn",
        "bsStyle": "default",
        "children": "Log in with Facebook",
        "className": "btn btn-primary btn-block",
        "disabled": false,
        "onClick": [Function],
        "type": "submit",
      },
      "ref": null,
      "rendered": "Log in with Facebook",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn btn-primary btn-block"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Log in with Facebook
        </Button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Log in with Facebook",
          "className": "btn btn-primary btn-block",
          "disabled": false,
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Log in with Facebook",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
